default/python/AI/turn_state.py

```diff
@@ -18,9 +24,61 @@ def __init__(self):
         self.__have_computronium = False
         self.__best_pilot_rating = 1e-8
         self.__medium_pilot_rating = 1e-8
+        self.__planets = {}
 
-    def cleanup(self):
+    def update(self):
         self.__init__()
+        universe = fo.getUniverse()
+        for pid in universe.planetIDs:
+            planet = universe.getPlanet(pid)
+            self.__planets[pid] = PlanetInfo(pid, planet.speciesName, planet.owner, planet.systemID)
+
+    def get_empire_species_systems(self):
+        """
+        Return dict from system id to planet ids of empire with species.
```

Comment:
|  The summary line may be on the same line as the opening quotes or on the next line. 

I prefer to put it in next line for multistring comments